File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.cpp
class Solution {
public:
    int subarraySum(vector<int>& nums, int k) {
        map<int,int>mpp;
        int n=nums.size();
        int cnt=0;
        vector<int>pref(n,0);
        pref[0]=nums[0];
        for(int i=1;i<n;i++){
            pref[i]=pref[i-1]+nums[i];
         }
         mpp[0]=1;
        for(int i=0;i<n;i++){
            // pref[i]=pref[i-1]+nums[i];
            if(mpp.find(pref[i]-k)!=mpp.end()){
                cnt+=mpp[pref[i]-k];
            }
            mpp[pref[i]]++;
        }
        return cnt;
    }
};

File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.py
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        mpp={}
        cnt=0
        mpp[0]=1
        s=0
        for i in range(len(nums)):
            s+=nums[i]
            if s-k in mpp:
                cnt+=mpp[s-k]
            mpp[s]=mpp.get(s,0)+1
        return cnt